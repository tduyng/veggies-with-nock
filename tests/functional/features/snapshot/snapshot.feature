@snapshot
Feature: Using snapshot step definitions

    Scenario: Snapshot testing on an API
        Given I mock a POST http call to forward request body for path /users/yaml
        And set request json body from file2
        When I POST http://fake.io/users/yaml
        Then response body should match snapshot
        
    Scenario: Snapshot testing on an API for GET request
        Given I mock a GET http call to forward request body for path /users/yaml
        When I GET http://fake.io/users/yaml
        Then response status code should be 200

    Scenario: Snapshot testing on an API with json file mock
        Given I mock a POST http call to forward request body for path /users/json
        And set request json body from file2
        When I POST http://fake.io/users/json
        Then response status code should be 200
        And response body should match snapshot

    Scenario: Snapshot testing on an API with json file mock for GET request
        Given I mock a GET http call to forward request body for path /users/json
        When I GET http://fake.io/users/json
        Then response status code should be 200

    Scenario: Snapshot testing on cli
        When I run command echo test
        Then exit code should be 0
        And stdout output should match snapshot

    Scenario: Snapshot testing on files
        Given I set cwd to tests/functional/features/snapshot/fixtures
        Then file file.yaml should match snapshot

    Scenario: Snapshot testing on a json API
        Given I mock a POST http call to forward request body for path /users/yaml
        And set request json body from file
        When I POST http://fake.io/users/yaml
        Then response json body should match snapshot
            | field      | matcher | value   |
            | id         | type    | number  |
            | first_name | equals  | Raphaël |
            | last_name  | equals  | Benitte |
            | gender     | equals  | male    |

    Scenario: Testing on a json API without snapshot
        Given I mock a POST http call to forward request body for path /users/yaml
        And set request json body from file
        When I POST http://fake.io/users/yaml
        Then json response should match
            | field      | matcher | value   |
            | id         | type    | number  |
            | first_name | equals  | Raphaël |
            | last_name  | equals  | Benitte |
            | gender     | equals  | male    |

    Scenario: Snapshot testing on a CLI with json output
        When I run command echo {"gender": "male", "name": "john"}
        Then exit code should be 0
        And stdout json output should match snapshot
            | field  | matcher | value |
            | gender | equals  | male  |
            | name   | equal   | john  |

    Scenario: Snapshot testing on a json file
        Given I set cwd to tests/functional/features/snapshot/fixtures
        Then json file file2.json content should match snapshot
            | field  | matcher | value  |
            | gender | type    | string |
            | id     | type    | number |
