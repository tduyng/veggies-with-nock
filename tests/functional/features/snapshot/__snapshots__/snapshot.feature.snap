

exports[`Snapshot testing on a CLI with json output 1.1`] = `Object {
  "gender": "equals(male)",
  "name": "equal(john)",
}`;

exports[`Snapshot testing on a CLI with json output 1.17`] = `Object {
  "gender": "equals(male)",
  "name": "equal(john)",
}`;

exports[`Snapshot testing on a json API 1.1`] = `Object {
  "first_name": "equals(Raphaël)",
  "gender": "equals(male)",
  "id": "type(number)",
  "last_name": "equals(Benitte)",
}`;

exports[`Snapshot testing on a json API 1.16`] = `Object {
  "first_name": "equals(Raphaël)",
  "gender": "equals(male)",
  "id": "type(number)",
  "last_name": "equals(Benitte)",
}`;

exports[`Snapshot testing on a json file 1.1`] = `Object {
  "first_name": "Raphaël",
  "gender": "type(string)",
  "id": "type(number)",
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on a json file 1.18`] = `Object {
  "first_name": "Raphaël",
  "gender": "type(string)",
  "id": "type(number)",
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on an API 1.1`] = `Object {
  "first_name": "Raphaël",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on an API 1.12`] = `Object {
  "first_name": "Raphaël",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on an API with json file mock 1.1`] = `Object {
  "first_name": "Raphaël",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on an API with json file mock 1.13`] = `Object {
  "first_name": "Raphaël",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on cli 1.1`] = `"test
"`;

exports[`Snapshot testing on cli 1.14`] = `"test
"`;

exports[`Snapshot testing on files 1.1`] = `"id:         1
first_name: Raphaël
last_name:  Benitte
gender:     male
"`;

exports[`Snapshot testing on files 1.15`] = `"id:         1
first_name: Raphaël
last_name:  Benitte
gender:     male
"`;
